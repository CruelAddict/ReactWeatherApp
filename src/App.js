import React from 'react';
import FavoritesList from './containers/FavoritesList'
import MainWeatherPanel from './containers/MainWeatherPanel'
import setGeoStatus from './modules/geolocationHandler'
import {connect} from 'react-redux'
import Actions from "./modules/actions";
import {NotificationContainer, NotificationManager} from 'react-notifications'

const mapStateToProps = state => ({
    permissionGranted: state.geo.permissionGranted,
    coordinates: state.geo.coordinates
});

const mapDispatchToProps = dispatch => ({
    createFavorite: item => dispatch(Actions.createFavorite(item)),
    fetchFavorite: item => dispatch(Actions.fetchFavorite(item))
});


class App extends React.Component {
    constructor(props) {
        super();
        this.state = {
            coordinates: props.coordinates
        }
    }

    fetchFavorites = async () => {
        fetch('http://127.0.0.1:3000/favourites').then(async response => {
            if (response.ok) {
                let items = await response.json();
                items.map(item => {this.props.fetchFavorite(item.name)})
            } else {
                NotificationManager.error('Не удалось загрузить список избранных городов', 'Сервер не отвечает')
            }
        }, response => {
            NotificationManager.error('Не удалось загрузить список избранных городов', 'Сервер не отвечает')
        })

    };


    componentDidMount() {
        this.fetchFavorites();
        setGeoStatus();
    }

    render() {
        return this.props.permissionGranted === null ? null
            :
            <div id={"app"}>
                <MainWeatherPanel coordinates={this.props.coordinates}/>
                <FavoritesList/>
                <NotificationContainer/>
            </div>
    };
}


export default connect(mapStateToProps, mapDispatchToProps)(App);
