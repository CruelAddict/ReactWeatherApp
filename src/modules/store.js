import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducer from './rootReducer'
import store from "../index";

const savedItems = localStorage.getItem('favorites') ? JSON.parse(localStorage.getItem('favorites')).map(value => {
        return {
            ...value,
            isLoading: true
        };
    }) : [];

const initialState = {
    favorites: {
        items: savedItems
    },
    geo: {
        permissionGranted: null,
        coordinates: null
    }
};

export default () => {
    return createStore(
        reducer,
        initialState,
        applyMiddleware(thunk)
    )
}
