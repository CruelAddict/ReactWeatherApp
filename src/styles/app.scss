body {
  margin: 0;
  padding: 0;
}

#app {
  display: grid;
  grid-template-columns: 1fr;
  width: 77%;
  margin: 0 auto;
  min-width: 750px;
}

button:focus {
  outline:0;
}

input:focus {
  outline: 1px solid grey;
}

button {
  cursor: pointer;
}
