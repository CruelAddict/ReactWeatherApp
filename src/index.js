import React from 'react';
import ReactDOM from 'react-dom';
import {AppConnected} from './App';
import {Provider} from 'react-redux';
import store from './modules/store';
import './styles/main.scss';

store.subscribe(() => {
    localStorage.setItem('favorites', JSON.stringify(store.getState().favorites.items.map(value => {
        let valueClone = {...value};
        delete valueClone['weatherObj'];
        console.log(valueClone);
        return valueClone;
    })))
});

ReactDOM.render(
    <Provider store={store}>
        <AppConnected/>
    </Provider>,
    document.getElementById('root'));

export default store
