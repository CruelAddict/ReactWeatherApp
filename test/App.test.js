import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as Enzyme from "enzyme";
Enzyme.configure({ adapter: new Adapter() });

import {App} from '../src/app';
import store from '../src/modules/store'



describe ('App', () => {

    it('Renders successfully', () => {
        const app = shallow(
            <App
                store={store}
            />
        );
        expect(app).toMatchSnapshot()
    });

});
