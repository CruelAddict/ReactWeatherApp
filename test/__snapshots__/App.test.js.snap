// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders successfully 1`] = `
<div
  id="app"
>
  <Connect(MainWeatherPanel) />
  <Connect(FavoritesList) />
  <NotificationContainer
    enterTimeout={400}
    leaveTimeout={400}
  />
</div>
`;
